python/foglamp/services/core/api/configuration.py

```diff
@@ -439,3 +460,63 @@
         raise web.HTTPNotFound(reason=str(ex))
 
     return web.json_response({"message": "Parent-child relationship for the parent-{} is deleted".format(category_name)})
+
+
+async def upload_script(request):
+    """ Upload script for a given config item
+
+    :Example:
+            curl -F "script=@filename.py" http://localhost:8081/foglamp/category/{category_name}/{config_item}/upload
+    """
+    category_name = request.match_info.get('category_name', None)
+    config_item = request.match_info.get('config_item', None)
+
+    category_name = urllib.parse.unquote(category_name) if category_name is not None else None
+    config_item = urllib.parse.unquote(config_item) if config_item is not None else None
+    cf_mgr = ConfigurationManager(connect.get_storage_async())
+    category_item = await cf_mgr.get_category_item(category_name, config_item)
+    if category_item is None:
+        raise web.HTTPNotFound(reason="No such Category item found for {}".format(config_item))
+
+    data = await request.post()
+
+    # contains the name of the file in string format
+    script_file = data.get('script')
+    if not script_file:
+        raise web.HTTPBadRequest(reason="Script file is missing")
+
+    # TODO: For the time being accepted extension is '.py'
+    script_filename = script_file.filename
+    if not script_filename.endswith('.py'):
```

Comment:
should we have a list of acceptable extensions on top of this module? Not sure if we want  to allow  more extensions?

Reply:
We will want to support more extensions, although at this level the extension is not relevant, it is just a file we are uploading. If the file has an extension on the machine we upload from it should have the same extension when save in the FogLAMP server.